packages/vm/src/cache.rs

```diff
@@ -71,6 +71,7 @@ pub struct Cache<A: BackendApi, S: Storage, Q: Querier> {
     type_api: PhantomData<A>,
     type_storage: PhantomData<S>,
     type_querier: PhantomData<Q>,
+    m: Mutex<()>,
```

Comment:
Let's call this `instantiation_lock`, because this is what it does: one instantiation at a time.

Is there a simpler lock type that locks like `Mutex` but does not hold data?

Reply:
Not that I know of. And not in [sync](https://doc.rust-lang.org/std/sync/index.html#structs) at least. Rust has this policy of "lock data, not code" (which I don't agree with, btw).

We can always do `type Lock Muxtex<()>` and I for one will sleep well at night with it.

Reply:
Alright, thanks for checking. Let's stick with `Mutex<()>` then